### DISCUSSION TITLE
训练的loss很大

### DISCUSSION THREAD

**Discussion by shinerdeng** [QwenLM/Qwen3-ASR#173] [General]
训练的jsonl文件如下：
{"audio": "asr_audios_四川话/eab581e4_405115.wav", "text": "language Chinese<asr_text>有点流眼水，但没啥多分泌物。一照光就特难受。"}
{"audio": "asr_audios_四川话/eab581e4_405115.wav", "text": "language Chinese<asr_text>之前在外头做过检查是吧？"}
.....
音频是清晰的，用原始模型可以推理正确。训练前手动打印一次loss输出很大loss=12点几。后续开始训练后loss从200多降到100多。训练完降低到0.8。用外部数据测试比原始的效果差。请问可能是哪里的原因呢？  训练数据大概都是10秒左右，训练完推理传1个几分钟的音频只识别到几个字，原始模型能全部识别。是要混合不同长度的音频去训练吗？稍微长一点显存就oom了，有什么技巧吗?

# 测试第一个 batch 的数据
    batch = next(iter(trainer.get_train_dataloader()))
    batch = trainer._prepare_inputs(batch)
    print("Labels shape:", batch["labels"].shape)
    print("Non -100 count per sample:", [(batch["labels"][i] != -100).sum().item() for i in range(batch["labels"].shape[0])])
    print("Labels unique values:", torch.unique(batch["labels"]))
    # 手动前向计算 loss
    with torch.no_grad():
        outputs = model(**batch)
        print("Initial loss from model.forward:", outputs.loss.item())
